### PR TITLE
Adding Odroid N2

### PR DESCRIPTION
**Pull request by paulcrawford**
Here is the second one.

### CODE REVIEW

File: src/board.py [hunk 1]
```diff
@@ -85,9 +85,12 @@
 elif board_id == ap_board.CORAL_EDGE_TPU_DEV:
     from adafruit_blinka.board.coral_edge_tpu import *
 
-elif board_id == ap_board.ODROID_C2:
-    from adafruit_blinka.board.odroidc2 import *
+#elif board_id == ap_board.ODROID_C2:
+    #from adafruit_blinka.board.odroidc2 import *
```
Comment (makermelissa) [inline src/board.py]: Is there a reason this was commented out?

File: src/adafruit_blinka/board/odroidn2.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
+"""Pin definitions for the Odroid N2."""
+
+from adafruit_blinka.microcontroller.amlogic.s922x import pin
+
+#GPIOX_0 = pin.GPIO228
```
Comment (makermelissa) [inline src/adafruit_blinka/board/odroidn2.py]: Can we remove all the commented out lines and leave the ones relevant to the N2 only?

File: src/adafruit_blinka/microcontroller/amlogic/s922x/pin.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
+from adafruit_blinka.microcontroller.generic_linux.libgpiod_pin import Pin
+
+#GPIO122 = Pin((0, 0))
```
Comment (makermelissa) [inline src/adafruit_blinka/microcontroller/amlogic/s922x/pin.py]: Same here with commented out lines.

File: src/board.py [hunk 4]
```diff
@@ -85,9 +85,12 @@
...
 
+elif board_id == ap_board.ODROID_N2:
+   from adafruit_blinka.board.odroidn2 import *
+# error above
```
Comment (makermelissa) [inline src/board.py]: What does this mean?

**Comment by makermelissa** [comment]
You can view Travis errors here: https://travis-ci.com/adafruit/Adafruit_Blinka/builds/122026302

**Comment by paulcrawford** [comment]
> On Aug 5, 2019, at 17:51, Melissa LeBlanc-Williams <notifications@github.com> wrote:
> 
> @makermelissa requested changes on this pull request.
> 
> Mostly removing commented out code and questions above about specific comments.
> 
> In src/board.py <https://github.com/adafruit/Adafruit_Blinka/pull/146#discussion_r310806314>:
> 
> > @@ -85,9 +85,12 @@
>  elif board_id == ap_board.CORAL_EDGE_TPU_DEV:
>      from adafruit_blinka.board.coral_edge_tpu import *
>  
> -elif board_id == ap_board.ODROID_C2:
> -    from adafruit_blinka.board.odroidc2 import *
> +#elif board_id == ap_board.ODROID_C2:
> +    #from adafruit_blinka.board.odroidc2 import *
>  
I was having trouble with getting an attribute for the N2 so I commented out the C2 to see what would have. It can clearly be deleted.
> Is there a reason this was commented out?
> 
> In src/adafruit_blinka/board/odroidn2.py <https://github.com/adafruit/Adafruit_Blinka/pull/146#discussion_r310806558>:
> 
> > @@ -0,0 +1,154 @@
> +"""Pin definitions for the Odroid N2."""
> +
> +from adafruit_blinka.microcontroller.amlogic.s922x import pin
> +
> +#GPIOX_0 = pin.GPIO228
> Can we remove all the commented out lines and leave the ones relevant to the N2 only?
> 
> 
Yes the commented lines were only to show you how I came up with the new code. If we are confident that all is well than you can certainly delete the commented lines.
> 
> In src/adafruit_blinka/microcontroller/amlogic/s922x/pin.py <https://github.com/adafruit/Adafruit_Blinka/pull/146#discussion_r310806647>:
> 
> > @@ -0,0 +1,211 @@
> +from adafruit_blinka.microcontroller.generic_linux.libgpiod_pin import Pin
> +
> +#GPIO122 = Pin((0, 0))
> Same here with commented out lines.
> 
> 
Same as above.
> 
> In src/board.py <https://github.com/adafruit/Adafruit_Blinka/pull/146#discussion_r310806778>:
> 
> >  
> +elif board_id == ap_board.ODROID_N2:
> +   from adafruit_blinka.board.odroidn2 import *
> +# error above
> What does this mean?
> 
> 
This was a comment on where the error was on Travis. Once I fixed the platform detect I put in the comment so I could commit Blinka without changing the actual code. My Travis only runs on Commit. Maybe there is a way to run Travis without making changes. The comment can, of course, be deleted.

File: src/board.py (hunk 1 above)
Comment (makermelissa) [reply src/board.py]: I meant it seems like commenting this out would break it for the Odroid C2 board.

**Comment by paulcrawford** [comment]
Yes I got this error all of the time when I was developing the code. The problem is that although the ODROID_N2 constant is in the file on PlatformDetect it is not set up as attribute of the module. When I installed both Blinka and PlatformDetect from my personal GitHub account and then checked with dir the N2 attribute was there:

Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import adafruit_platformdetect.board
>>> dir(adafruit_platformdetect.board)
['BEAGLEBONE', 'BEAGLEBONE_AIR', 'BEAGLEBONE_BLACK', 'BEAGLEBONE_BLACK_INDUSTRIAL', 'BEAGLEBONE_BLACK_WIRELESS', 'BEAGLEBONE_BLUE', 'BEAGLEBONE_ENHANCED', 'BEAGLEBONE_GREEN', 'BEAGLEBONE_GREEN_WIRELESS', 'BEAGLEBONE_POCKETBEAGLE', 'BEAGLEBONE_POCKETBONE', 'BEAGLEBONE_USOMIQ', 'BEAGLELOGIC_STANDALONE', 'Board', 'CORAL_EDGE_TPU_DEV', 'DRAGONBOARD_410C', 'FEATHER_HUZZAH', 'FEATHER_M0_EXPRESS', 'FTDI_FT232H', 'GENERIC_LINUX_PC', 'GIANT_BOARD', 'JETSON_NANO', 'JETSON_TX1', 'JETSON_TX2', 'JETSON_XAVIER', 'NODEMCU', 'ODROID_C1', 'ODROID_C1_PLUS', 'ODROID_C2', 'ODROID_N2', 'ORANGE_PI_PC', 'ORANGE_PI_R1', 'OSD3358_DEV_BOARD', 'OSD3358_SM_RED', 'PYBOARD', 'RASPBERRY_PI_2B', 'RASPBERRY_PI_3A_PLUS', 'RASPBERRY_PI_3B', 'RASPBERRY_PI_3B_PLUS', 'RASPBERRY_PI_4B', 'RASPBERRY_PI_A', 'RASPBERRY_PI_A_PLUS', 'RASPBERRY_PI_B_PLUS', 'RASPBERRY_PI_B_REV1', 'RASPBERRY_PI_B_REV2', 'RASPBERRY_PI_CM1', 'RASPBERRY_PI_CM3', 'RASPBERRY_PI_CM3_PLUS', 'RASPBERRY_PI_ZERO', 'RASPBERRY_PI_ZERO_W', 'SIFIVE_UNLEASHED', '_BEAGLEBONE_BOARD_IDS', '_BEAGLEBONE_IDS', '_CORAL_IDS', '_JETSON_IDS', '_LINARO_96BOARDS_IDS', '_ODROID_40_PIN_IDS', '_ORANGE_PI_IDS', '_PI_REV_CODES', '_RASPBERRY_PI_40_PIN_IDS', '_RASPBERRY_PI_CM_IDS', '_SIFIVE_IDS', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'ap_chip', 'os']

**Comment by paulcrawford** [comment]
Yes it would so it should be back in.

**Comment by makermelissa** [comment]
Makes sense on the Travis error. We may have to merge and release on Platform Detect and circle back to this.

**Comment by paulcrawford** [comment]
Just for completeness the following are the responses on my C2:

root@no:/home/co# cat /proc/device-tree/model
ODROID-C2

root@no:/home/co# cat /proc/device-tree/compatible
amlogic, Gxbb

I did not try installing Blinka and PlatformDetect on it as yet.

**Comment by ladyada** [comment]
man that is minimal :D ok thanks for the details!

**Comment by makermelissa** [comment]
Ok, now that Platform detect was added and release, I'll wait a few minutes and then restart travis.

**Comment by makermelissa** [comment]
@paulcrawford, I made the changes I requested, reorganized all the microcontroller pins and removed some duplicates I found from doing that. Please test again. Thanks!

**Comment by paulcrawford** [comment]
I will also update both repositories from adafruit/… and recheck.

Done and checked. Works perfectly.

**Comment by makermelissa** [comment]
Thanks!

**Pull request merged**